### ISSUE TITLE
Rendezvous-Point Routing

### ISSUE THREAD

**Issue by matsjj** [blockchain/thunder#9]
Currently we are directly routing towards a public key, which can easily fail when we don't have a route to a specific wallet in our database. 

Routing towards a 24/7 online node as a rendezvous point however is more likely to succeed. 

To implement it, the receiver has to choose a nearby node it is having a channel with and create a partial OnionObject for the route from that node to himself. Furthermore, that OnionObject needs to be included in the payment request data, such that the sender can include it into his calculation of the route. 

For now, we can keep it at a one-node distance from the receiver.

**Comment by Giszmo** [comment]
Your comment implies there is already onion routing in place? Would probably warrant another checked checkbox "[✓] privacy thanks to onion routing", no?

Your feature would allow the recipient to pick up to several hops and would not be limited to dictating his entry node, which would be the case if some email-address-like scheme was implemented, where Tony@bitpay pays to Nic@blockchain.

**Comment by matsjj** [comment]
Yes it is in place already, see `LNOnionHelperImpl` for how it's implemented. Might want to highlight it, thanks for the feedback. ;)

Ah right, yes, they receiver can indeed chose an arbitrary amount of nodes and send it all over to the sender, who can then send to the node which is cheapest/fastest for it. It might give away information about the receiver though.
